### PR TITLE
Refactor `PhpMyAdmin\Plugins::getPlugins` method

### PR DESCRIPTION
**Pull request by MauricioFauth**

### CODE REVIEW

File: libraries/classes/Plugins.php [hunk 1]
```diff
@@ -96,97 +98,66 @@ public static function getExport(string $type, bool $singleTable): array
...
-        global $skip_import;
-
-        $GLOBALS['plugin_param'] = $plugin_param;
-
-        $fullFsPathPluginDir = ROOT_PATH . $plugins_dir;
-
-        $handle = @opendir($fullFsPathPluginDir);
-        if (! $handle) {
+        try {
+            $files = new FilesystemIterator(ROOT_PATH . 'libraries/classes/Plugins/' . $type);
```
Comment (williamdes) [inline libraries/classes/Plugins.php:126]: I think you could use the folderopen way, we have a FileListing class but it has some useless regex code in this case
Anyway, good for now

It's just not the most performant method

File: libraries/classes/Plugins/Export/ExportXml.php [hunk 2]
```diff
@@ -616,4 +603,12 @@ private function setTables(array $tables)
     {
         $this->tables = $tables;
     }
+
+    public function isAvailable(): bool
+    {
+        global $db;
+
+        // Can't do server export.
```
Comment (williamdes) [inline libraries/classes/Plugins/Export/ExportXml.php:611]:
```suggestion
        // Can't do a server export without a database name.
```

File: composer.json [hunk 3]
```diff
@@ -58,6 +58,7 @@
         "symfony/expression-language": "^5.2.3",
         "symfony/polyfill-ctype": "^1.17.0",
         "symfony/polyfill-mbstring": "^1.17.0",
+        "symfony/polyfill-php80": "^1.16",
```
Comment (williamdes) [inline composer.json:61]: why is it needed ?

**Review by williamdes — APPROVED** [review]
Very good work on removing this global

File: composer.json (hunk 3 above)
Comment (mauriciofauth) [reply composer.json:61]: I used it for `str_starts_with`. We already have this package installed, I'm just using it.

```
composer why symfony/polyfill-php80
```

Comment (williamdes) [reply composer.json:61]: Oh okay, if you plan to use it in some other places I agree 100% to use the polyfill

File: libraries/classes/Plugins.php (hunk 1 above)
Comment (mauriciofauth) [reply libraries/classes/Plugins.php:126]: Using the Iterator was slower in my tests, but the difference was so small that it was insignificant.

File: libraries/classes/Plugins/Export/ExportXml.php (hunk 2 above)
Comment (mauriciofauth) [reply libraries/classes/Plugins/Export/ExportXml.php:611]: XML export is only available when exporting a database or a table, not when exporting the server.

**Pull request merged**